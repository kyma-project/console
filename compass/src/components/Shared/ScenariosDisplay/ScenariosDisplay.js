import React from 'react';
import PropTypes from 'prop-types';
import { Token } from '@kyma-project/react-components';
import { EMPTY_TEXT_PLACEHOLDER } from '../../../shared/constants';

ScenariosDisplay.propTypes = {
  scenarios: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  className: PropTypes.string,
  emptyPlaceholder: PropTypes.string,
};

ScenariosDisplay.defaultProps = {
  emptyPlaceholder: EMPTY_TEXT_PLACEHOLDER,
};

export default function ScenariosDisplay({
  scenarios,
  className,
  emptyPlaceholder,
}) {
  if (!scenarios.length) {
    return <span className={className}>{emptyPlaceholder}</span>;
  }

  return (
    <span className={className}>
      {scenarios.map(scenario => (
        <Token
          key={scenario}
          className="y-fd-token y-fd-token--no-button fd-has-margin-right-tiny"
        >
          {scenario}
        </Token>
      ))}
    </span>
  );
}
