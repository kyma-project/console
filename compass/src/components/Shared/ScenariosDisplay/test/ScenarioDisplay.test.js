import React from 'react';
import renderer from 'react-test-renderer';
import ScenariosDisplay from '../ScenariosDisplay';

describe('ScenariosDisplay', () => {
  it('Renders scenario list', () => {
    const component = renderer.create(
      <ScenariosDisplay scenarios={['a', 'b', 'c']} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders default placeholder when there are no scenarios', () => {
    const component = renderer.create(<ScenariosDisplay scenarios={[]} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders custom placeholder when there are no scenarios', () => {
    const component = renderer.create(
      <ScenariosDisplay scenarios={[]} emptyPlaceholder={'No scenarios'} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders custom class name', () => {
    const component = renderer.create(
      <ScenariosDisplay scenarios={['a', 'b', 'c']} className="class-name" />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
