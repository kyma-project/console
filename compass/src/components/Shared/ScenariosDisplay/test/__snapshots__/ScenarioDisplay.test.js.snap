// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScenariosDisplay Renders custom class name 1`] = `
<span
  className="class-name"
>
  <span
    className="fd-token y-fd-token y-fd-token--no-button fd-has-margin-right-tiny sc-jbKcbu hnJCGk"
    role="button"
  >
    a
  </span>
  <span
    className="fd-token y-fd-token y-fd-token--no-button fd-has-margin-right-tiny sc-jbKcbu hnJCGk"
    role="button"
  >
    b
  </span>
  <span
    className="fd-token y-fd-token y-fd-token--no-button fd-has-margin-right-tiny sc-jbKcbu hnJCGk"
    role="button"
  >
    c
  </span>
</span>
`;

exports[`ScenariosDisplay Renders custom placeholder when there are no scenarios 1`] = `
<span>
  No scenarios
</span>
`;

exports[`ScenariosDisplay Renders default placeholder when there are no scenarios 1`] = `
<span>
  -
</span>
`;

exports[`ScenariosDisplay Renders scenario list 1`] = `
<span>
  <span
    className="fd-token y-fd-token y-fd-token--no-button fd-has-margin-right-tiny sc-jbKcbu hnJCGk"
    role="button"
  >
    a
  </span>
  <span
    className="fd-token y-fd-token y-fd-token--no-button fd-has-margin-right-tiny sc-jbKcbu hnJCGk"
    role="button"
  >
    b
  </span>
  <span
    className="fd-token y-fd-token y-fd-token--no-button fd-has-margin-right-tiny sc-jbKcbu hnJCGk"
    role="button"
  >
    c
  </span>
</span>
`;
